### PR TITLE
[Chore] Update Project for Xcode 14

### PR DESCRIPTION
**Pull request by blyscuit**
## What happened

Update Project for Xcode 14.
 
## Insight

- Some library will result in additional `bundle` and have to disable signing via `podfile`. https://stackoverflow.com/a/74129238/3736373
- Remove `bitcode` from `BuildManager` because bitcode will fail TestFlight upload.
- `macOS-latest` is [macOS-11](https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md) and does not have Xcode 14. We should change to `macOS-latest` when Actions update the default https://github.com/actions/runner-images/issues/6384.
 
## Proof Of Work

https://github.com/nimblehq/ios-templates/actions/runs/3416880487

### CODE REVIEW

**Review by ducbm051291 — APPROVED** [review]
LGTM 👍

**Review by phongvhd93 — APPROVED** [review]

File: Podfile [hunk 1]
```diff
@@ -53,6 +53,12 @@ post_install do |installer|
     target.build_configurations.each do |config|
       config.build_settings.delete 'IPHONEOS_DEPLOYMENT_TARGET'
       config.build_settings['ENABLE_BITCODE'] = 'NO'
+      # Fix some pods creating bunndle, asking for signing
```
Comment (nkhanh44) [inline Podfile:56]:
```suggestion
      # Fix some pods creating bundle, asking for signing
```

**Review by nkhanh44 — APPROVED** [review]
Looks great 🚀 . Just face this issue too.

File: .github/workflows/test_upload_build_to_test_flight.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ on:
 jobs:    
   build:
     name: Build
-    runs-on: macOS-latest
+    runs-on: macOS-12
```
Comment (minhnimble) [inline .github/workflows/test_upload_build_to_test_flight.yml:17]: Just curious why do we need to make this change to run specifically on `macOS-12` only? Isn't the fix for supporting the latest environment? 🤔

File: Podfile (hunk 1 above)
Comment (blyscuit) [reply Podfile:56]: Nice catch, time to change my keyboard switches 🙈

File: .github/workflows/test_upload_build_to_test_flight.yml (hunk 2 above)
Comment (blyscuit) [reply .github/workflows/test_upload_build_to_test_flight.yml:17]: macOS-lastest is [macOS-11](https://github.com/actions/runner-images/blob/main/images/macos/macos-11-Readme.md) and does not have Xcode 14. We should change to macOS-lastest when Actions update the default [macOS version ](https://github.com/actions/runner-images/issues/6384).

Comment (minhnimble) [reply .github/workflows/test_upload_build_to_test_flight.yml:17]: @blyscuit that is a great insight! Then I think you can also include it as a part of the `Insight` section in the PR description for later action also 👍 Rest LGTM 🚀

**Review by minhnimble — APPROVED** [review]

**Review by vnntsu — APPROVED** [review]
LGTM! 💪

**Review by Shayokh144 — APPROVED** [review]

**Comment by vnntsu** [comment]
@blyscuit Could you help me to reopen this PR with the base is `develop`?

**Pull request closed without merging**